### PR TITLE
feat(DNS): Add per stage own DNS APEX domain example

### PR DESCRIPTION
**Pull request by flochaz**
# Context

Current AWS Bootstrap Kit design only propose management of DNS across stages with a single account (leveraging subdomains). This PR is a simple example showing how it would be possible to have a dedicated domain per stage and how to deploy this configuration.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by armenr** [comment]
+1 to seeing this get merged!

File: source/per-stage-dns/README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    dev-mydomain.com. 300 IN NS ns-890.awsdns-12.net.
+    ...
+    ```
+
+
+### 3. Deploy to stages through pipeline
+
+1. Build and Deploy the pipeline
+    ```
+    npm run cdk deploy DNSInfrastructurePipelineStack --  --profile mvpv3-cicd
```
Comment (alex-m-aws) [inline source/per-stage-dns/README.md:83]: mvpv3-cicd is your profile name, but will users also have this profile name? If not, could replace to --profile=<CLI_CICD_PROFILE> or --profile=cicd

File: source/per-stage-dns/README.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            this,
+            "Certificate",
+            {
+                hostedZone: stageHostedZone,
+                domainName: stageHostedZone.zoneName,
+                region: "us-east-1",
+                validation: CertificateValidation.fromDns(stageHostedZone),
+            }
+        );
+    ...
```
Comment (alex-m-aws) [inline source/per-stage-dns/README.md:139]: missing closing ```

File: source/per-stage-dns/lib/cicd-pipeline-stack.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            }),
+            cloudAssemblyArtifact: cloudAssemblyArtifact,
+        });
+
+
+        new CfnOutput(this, "PipelineConsoleUrl", {
+            value: `https://${Stack.of(this).region}.console.aws.amazon.com/codesuite/codepipeline/pipelines/${pipeline.codePipeline.pipelineName}/view?region=${Stack.of(this).region}`,
+        });
+
+        const AWS_PROFILE = "cicd";
```
Comment (alex-m-aws) [inline source/per-stage-dns/lib/cicd-pipeline-stack.ts:67]: Is it a good idea to hardcode a CLI profile name?

**Review by alex-m-aws — CHANGES_REQUESTED** [review]
Main issue is that there are some assumptions about the user's CLI profile setup. If these are non-issues LMK in replies

File: source/per-stage-dns/lib/cicd-pipeline-stack.ts (hunk 3 above)
Comment (flochaz) [reply source/per-stage-dns/lib/cicd-pipeline-stack.ts:67]: Yeah that the way I can log to the account the appropriate permission ... this profile should having created through AWS Bootstrap Kit dev process

**Review by alex-m-aws — APPROVED** [review]

**Pull request merged**